### PR TITLE
Use mutable data structures in CfgDominator calculation.

### PR DESCRIPTION
**Pull request by ml86**
Only a small gain of a few seconds for very large CPGs but since it is
already changed we might as well take it.

### CODE REVIEW

File: semanticcpg/src/main/scala/io/shiftleft/semanticcpg/passes/cfgdominator/CfgDominator.scala [hunk 1]
```diff
@@ -87,10 +89,10 @@ class CfgDominator[NodeType](adapter: CfgAdapter[NodeType]) {
     finger1
   }
 
-  private def createPostOrderNumbering(cfgEntry: NodeType): Map[NodeType, Int] = {
+  private def createPostOrderNumbering(cfgEntry: NodeType): mutable.Map[NodeType, Int] = {
     var stack = (cfgEntry, adapter.successors(cfgEntry).iterator) :: Nil
-    var visited = Set.empty[NodeType]
-    var numbering = Map.empty[NodeType, Int]
+    var visited = mutable.Set.empty[NodeType]
```
Comment (bbrehm) [inline semanticcpg/src/main/scala/io/shiftleft/semanticcpg/passes/cfgdominator/CfgDominator.scala:94]: If we're touching this code anyways, may as well `val visited = ...`

**Review by bbrehm — APPROVED** [review]
These changes are fine. If we find the opportunity, I'd like to move the DFO-Pass from codescience to codepropertygraph and merge it with this.

File: semanticcpg/src/main/scala/io/shiftleft/semanticcpg/passes/cfgdominator/CfgDominator.scala (hunk 1 above)
Comment (ml86) [reply semanticcpg/src/main/scala/io/shiftleft/semanticcpg/passes/cfgdominator/CfgDominator.scala]: Done.

**Comment by ml86** [comment]
Yes, reusing the DFO makes sense at some point.

**Pull request merged**